cql3/statements/create_keyspace_statement.cc

```diff
@@ -76,6 +76,14 @@ void create_keyspace_statement::validate(query_processor& qp, const service::cli
     } catch (const std::runtime_error& e) {
         throw exceptions::invalid_request_exception(e.what());
     }
+
+    const auto& tablets_opts = _attrs->get_map(ks_prop_defs::KW_TABLETS);
+    if (tablets_opts.has_value() && !_attrs->get_replication_strategy_class()->ends_with("NetworkTopologyStrategy")) {
```

Comment:
The correct place is `abstract_replication_strategy::validate_replication_strategy()`. It checks that the user-provided options are at all supported by the replication strategy. Apparently it doesn't work now, but why?

Reply:
From what I understand, `abstract_replication_strategy::validate_replication_strategy()` is validating options from strategy's POV, whereas we're interested in configuration provided to the CREATE KS statement. I just checked that `abstract_replication_strategy::validate_replication_strategy()` is indeed called when CREATE KS is executed, but it returns `true`, since all the strategy's options we're passing the the CREATE KS are legit. But the problem is "one level above" - not in the strategy options, but the KS options, namely - CREATE tablets KS cannot accept SimpleStrategy (or any other strategy, except for Network, for that matter). So this looks like the correct place to me.

Reply:
Yes, you're right. The `validate_replication_strategy()` currently only checks if the options from "WITH REPLICATION ..." suite the strategy itself.
I still think that it's the same function that should validate the rest of `replication_strategy_params params`, something like this should work

```c++
    if (params.initial_tablets.has_value() && !strategy.is_per_table()) {
        throw ...
    }
```

?

Reply:
I naively followed your advice and wow! What a mess you tricked me into! Now you have to review all of it ;)